core/referral/engine.go

```diff
@@ -138,6 +142,10 @@ func (e *Engine) CreateReferralSet(ctx context.Context, party types.PartyID, det
 		return ErrIsAlreadyAReferee(party)
 	}
 
+	if err := e.isPartyEligible(party); err != nil {
```

Comment:
I don't think this check is needed. We already have a spam check that won't allow a party with insufficient funds to createReferralSet.